pkg/kclient/pods.go

```diff
@@ -0,0 +1,118 @@
+package kclient
+
+import (
+	"io"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/pkg/errors"
+
+	// api resource types
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const (
+	// waitForPodTimeOut controls how long we should wait for a pod before giving up
+	waitForPodTimeOut = 240 * time.Second
+)
+
+// WaitAndGetPod block and waits until pod matching selector is in in Running state
+// desiredPhase cannot be PodFailed or PodUnknown
+func (c *Client) WaitAndGetPod(watchOptions metav1.ListOptions, desiredPhase corev1.PodPhase, waitMessage string) (*corev1.Pod, error) {
+	glog.V(4).Infof("Waiting for %s pod", watchOptions.LabelSelector)
+	s := log.Spinner(waitMessage)
+	defer s.End(false)
+
+	w, err := c.KubeClient.CoreV1().Pods(c.Namespace).Watch(watchOptions)
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to watch pod")
+	}
+	defer w.Stop()
+
+	podChannel := make(chan *corev1.Pod)
+	watchErrorChannel := make(chan error)
+
+	go func() {
+	loop:
+		for {
+			val, ok := <-w.ResultChan()
+			if !ok {
+				watchErrorChannel <- errors.New("watch channel was closed")
+				break loop
```

Comment:
can we `return` here instead of breaking?

Reply:
Sure. I've updated the goroutine to use `returns` instead of `breaks` in the loop.